src/Administration/Resources/app/administration/src/app/component/data-grid/sw-data-grid/index.js

```diff
@@ -769,6 +777,36 @@ export default {
             this.sort(column);
         },
 
+        onRowClick(event, item) {
+            if (!this.rowsClickable) {
+                return;
+            }
+
+            const target = event.target;
+
+            const blockedSelectors = [
+                '.sw-data-grid__cell--selection',
+                '.sw-data-grid__cell--actions',
+                '.sw-context-button',
+                'button',
+                'a',
+                'input',
+            ];
+
+            if (blockedSelectors.some((selector) => target.closest(selector))) {
+                return;
+            }
+
+            if (this.showSelection) {
+                const itemId = item[this.itemIdentifierProperty];
+                const isCurrentlySelected = this.isSelected(itemId);
+
+                this.selectItem(!isCurrentlySelected, item);
+            }
+
+            this.$emit('row-click', item);
```

Comment:
For me the big use case of this would be allowing to click a whole row to go to the detail page of something. Right now you often have to click the "title" in some specific column (with a rather small click area) 🎉 .

Using it for selection is also fine if the rows are selectable as you have implemented here.

So I guess the next step would be using this feature in most places where it would make sense? And I guess you already talked with our UI/UX department? 🤔